share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml

```diff
@@ -327,7 +327,8 @@ spack:
       - cd ${SPACK_CONCRETE_ENV_DIR}
       - spack env activate --without-view .
       - spack config add "config:install_tree:projections:${SPACK_JOB_SPEC_PKG_NAME}:'morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'"
-      - spack -d ci rebuild
+      - mkdir -p ${SPACK_ARTIFACTS_ROOT}/user_data
+      - spack -d ci rebuild > >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_out.txt) 2> >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_err.txt >&2)
```

Comment:
I'm gonna need some help understanding what's going on here WRT output redirection

Reply:
I want stdout and stderr to continue going to the terminal (or gitlab trace), but also to be sent to separate files.  The accepted answer on this SO [question](https://stackoverflow.com/questions/692000/how-do-i-write-stderr-to-a-file-while-using-tee-with-a-pipe) does a great job of breaking down the command.

Reply:
```
[spack -d ci rebuild]
   \stdout -> tee -> terminal_stdout (you see this on the gitlab job log)
                \--> pipeline_out.txt
   \stderr -> tee -> terminal_stdout (you see this on the gitlab job log) (note the >&2)
                \--> pipeline_err.txt
```